src/otk/transform.py

```diff
@@ -65,31 +65,37 @@ def resolve_dict(ctx: Context, state: State, tree: dict[str, Any]) -> Any:
         if isinstance(val, str):
             val = substitute_vars(ctx, val)
         if is_directive(key):
-            # Define, target, and version are done separately, they allow
-            # sibling elements thus they return the tree with their key set to their
-            # (resolved) value.
+
```

Comment:
(super nitpick) stray newline here (or is it intensional?)

Reply:
It was intentional to make the nested if blocks stand out:
```python
        if is_directive(key):

            if key.startswith(PREFIX_DEFINE):
                process_defines(ctx, state, val)
                continue

            if key == NAME_VERSION:
                continue

            if key.startswith(PREFIX_TARGET):
                continue
```